packages/useScript/src/useScript.ts

```diff
@@ -1,18 +1,54 @@
-import { useEffect, useState } from 'react'
+import { useEffect, useReducer } from 'react'
+
+enum ScriptStatus {
+  LOADED = 'loaded',
+  ERROR = 'error',
+}
+
+type Action = {
+  type: ScriptStatus
+}
+
+type State = {
+  ready: boolean
+  error?: boolean
+}
+
+const INITIAL_STATE: State = { ready: false, error: undefined }
+
+function scriptLoadReducer(state: State, action: Action): State {
+  switch (action.type) {
+    case ScriptStatus.LOADED:
+      return { ...INITIAL_STATE, ready: true }
+    case ScriptStatus.ERROR:
+      return { ...INITIAL_STATE, error: true }
+    default: {
+      throw new Error('Invalid action dispatched.')
+    }
+  }
+}
 
 /**
  * Hook to load an external script. Returns true once the script has finished loading.
  *
  * @param {string} url The external script to load
  * @return boolean True if the script has been loaded
  * */
-export default function useScript(url: string): boolean {
-  const [ready, setReady] = useState<boolean>(false)
+export default function useScript(url: string): [boolean, boolean?] {
+  const [{ ready, error }, dispatch] = useReducer(
```

Comment:
I think the `useReducer` can be replaced by a simpler `useState` and just saving the enum value of `ScriptStatus`. Right now the reducer just sets one of two `booleans.

```ts
const [status, setStatus] = useState<ScriptStatus>(ScriptStatus.INITIAL)
```